client/src/remote/__tests__/PluginsSpec.js

```diff
@@ -2,94 +2,85 @@
 
 import Plugins from '../Plugins';
 
-const stub = sinon.stub;
-
-const noop = () => {};
-
-
-describe('Plugins', function() {
-
-  let plugins;
-
-  beforeEach(function() {
-    delete window.plugins;
-
-    plugins = new Plugins({
-      plugins: {
-        getAll() {
-          return {
-            foo: {
-              name: 'foo',
-              script: 'foo.js'
-            },
-            bar: {
-              name: 'bar',
-              script: 'bar.js'
-            },
-            baz: {
-              name: 'baz',
-              style: 'baz.css'
-            }
-          };
+
+describe('plugins', function() {
+
+  describe('#loadAll', function() {
+
+    it('should load all plugins', async function() {
+      // given
+      const headStub = sinon.stub(document.head, 'appendChild')
+        .callsFake(({ onload }) => onload && onload());
+
+      const mockPlugins = {
+        foo: {
+          name: 'foo',
+          script: 'bar',
+          style: 'baz'
+        },
+        bar: {
+          name: 'bar',
+          script: 'foo',
+          style: 'baz'
         }
-      }
-    });
+      };
 
-    stub(plugins, '_loadStylePlugin').callsFake(noop);
-    stub(plugins, '_loadScriptPlugin').callsFake(({ name }) => {
-      if (!window.plugins) {
-        window.plugins = [];
-      }
-
-      window.plugins.push({
-        type: name,
-        plugin: {
-          __init__: [ name ],
-          [ name ]: [ 'type', noop ]
+      const plugins = new Plugins({
+        plugins: {
+          getAll() {
+            return mockPlugins;
+          }
         }
       });
 
-      Promise.resolve();
+      // when
+      await plugins.loadAll();
+
+      // then
+      const calls = headStub.getCalls();
+
+      expect(calls).to.have.lengthOf(4);
+
+      const args = calls.reduce((args, call) => [ ...args, ...call.args ], []);
+
+      expect(args.filter(({ tagName }) => tagName === 'LINK')).to.have.lengthOf(2);
+      expect(args.filter(({ tagName }) => tagName === 'SCRIPT')).to.have.lengthOf(2);
+
+      // teardown
+      headStub.restore();

```

Comment:
We must restore stubs in an after call, otherwise they won't be restored on test failure.

Reply:
Thanks for the feedback. I'll change this as you suggested.